applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py

```diff
@@ -58,12 +58,25 @@ def AddDofs(self):
         self._add_dynamic_dofs()
         self.print_on_rank_zero("::[ImplicitMechanicalSolver]:: ", "DOF's ADDED")
 
+    def InitializeSolutionStep(self):
+        # Using the base InitializeSolutionStep
+        super(ImplicitMechanicalSolver, self).InitializeSolutionStep()
+
+        # Some pre-processes may affect the system of equations, we rebuild the equation ids
+        if self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] == 1:
```

Comment:
```suggestion
        if self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] == 1 and self.main_model_part.ProcessInfo[StructuralMechanicsApplication.RESET_EQUATION_IDS]:
```
This way you can set it in your process and it will be disabled by default (ofc this requires a new variable)

Also wouldn't it be safer to set `SetDofSetIsInitializedFlag` of the B&S to false here, then also the dofset and the system-vectors & -matrices will be reinitialized? Otherwise I fear there could be some inconsistencies

@jcotela what do you think abt this?

Reply:
This variable actually exists or it is an example?

Reply:
was an example => a new one needs to be created

Reply:
OK